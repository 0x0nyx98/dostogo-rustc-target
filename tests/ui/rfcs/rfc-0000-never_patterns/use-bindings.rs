#![feature(never_patterns)]
#![allow(incomplete_features)]

#[derive(Copy, Clone)]
enum Void {}

fn main() {
    let res_void: Result<bool, Void> = Ok(true);

    let (Ok(x) | Err(!)) = res_void;
    println!("{x}");
    //~^ ERROR: used binding `x` is possibly-uninitialized
    let (Ok(x) | Err(!)) = &res_void;
    println!("{x}");
    //~^ ERROR: used binding `x` is possibly-uninitialized
    let (Err(!) | Ok(x)) = res_void;
    println!("{x}");
    //~^ ERROR: used binding `x` is possibly-uninitialized

    match res_void {
        Ok(x) | Err(!) => println!("{x}"),
        //~^ ERROR: used binding `x` is possibly-uninitialized
    }
    match res_void {
        Err(!) | Ok(x) => println!("{x}"),
        //~^ ERROR: used binding `x` is possibly-uninitialized
    }

    let res_res_void: Result<Result<bool, Void>, Void> = Ok(Ok(true));
    match res_res_void {
        Ok(Ok(x) | Err(!)) | Err(!) => println!("{x}"),
        //~^ ERROR: used binding `x` is possibly-uninitialized
    }
}
