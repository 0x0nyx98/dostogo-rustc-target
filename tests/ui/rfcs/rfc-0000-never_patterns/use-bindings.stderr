error[E0381]: used binding `x` is possibly-uninitialized
  --> $DIR/use-bindings.rs:11:15
   |
LL |     let (Ok(x) | Err(!)) = res_void;
   |             -
   |             |
   |             binding initialized here in some conditions
   |             binding declared here but left uninitialized
LL |     println!("{x}");
   |               ^^^ `x` used here but it is possibly-uninitialized
   |
   = note: this error originates in the macro `$crate::format_args_nl` which comes from the expansion of the macro `println` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0381]: used binding `x` is possibly-uninitialized
  --> $DIR/use-bindings.rs:14:15
   |
LL |     let (Ok(x) | Err(!)) = &res_void;
   |             -
   |             |
   |             binding initialized here in some conditions
   |             binding declared here but left uninitialized
LL |     println!("{x}");
   |               ^^^ `x` used here but it is possibly-uninitialized
   |
   = note: this error originates in the macro `$crate::format_args_nl` which comes from the expansion of the macro `println` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0381]: used binding `x` is possibly-uninitialized
  --> $DIR/use-bindings.rs:17:15
   |
LL |     let (Err(!) | Ok(x)) = res_void;
   |                      -
   |                      |
   |                      binding initialized here in some conditions
   |                      binding declared here but left uninitialized
LL |     println!("{x}");
   |               ^^^ `x` used here but it is possibly-uninitialized
   |
   = note: this error originates in the macro `$crate::format_args_nl` which comes from the expansion of the macro `println` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0381]: used binding `x` is possibly-uninitialized
  --> $DIR/use-bindings.rs:21:37
   |
LL |         Ok(x) | Err(!) => println!("{x}"),
   |            -                        ^^^ `x` used here but it is possibly-uninitialized
   |            |
   |            binding initialized here in some conditions
   |            binding declared here but left uninitialized
   |
   = note: this error originates in the macro `$crate::format_args_nl` which comes from the expansion of the macro `println` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0381]: used binding `x` is possibly-uninitialized
  --> $DIR/use-bindings.rs:25:37
   |
LL |         Err(!) | Ok(x) => println!("{x}"),
   |                     -               ^^^ `x` used here but it is possibly-uninitialized
   |                     |
   |                     binding initialized here in some conditions
   |                     binding declared here but left uninitialized
   |
   = note: this error originates in the macro `$crate::format_args_nl` which comes from the expansion of the macro `println` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0381]: used binding `x` is possibly-uninitialized
  --> $DIR/use-bindings.rs:31:50
   |
LL |         Ok(Ok(x) | Err(!)) | Err(!) => println!("{x}"),
   |               -                                  ^^^ `x` used here but it is possibly-uninitialized
   |               |
   |               binding initialized here in some conditions
   |               binding declared here but left uninitialized
   |
   = note: this error originates in the macro `$crate::format_args_nl` which comes from the expansion of the macro `println` (in Nightly builds, run with -Z macro-backtrace for more info)

error: aborting due to 6 previous errors

For more information about this error, try `rustc --explain E0381`.
